Slim/Menu/BrowseLibrary/Releases.pm

```diff
@@ -65,6 +67,21 @@ sub _releases {
 		# map to role's name for readability
 		$_->{role_ids} = join(',', map { Slim::Schema::Contributor->roleToType($_) } split(',', $_->{role_ids} || ''));
 
+		my $genreMatch = undef;
+		if ( !( $menuMode && $menuMode ne 'artists' && $menuRoles ) && $prefs->get('showComposerReleasesbyAlbum')==2 ) {
+			my $requestRef = [ 'genres', 0, MAX_ALBUMS, "album_id:".$_->{id} ];
+			my $request = Slim::Control::Request->new( undef, $requestRef );
+			$request->execute();
+			$log->error($request->getStatusText()) if $request->isStatusError();
+			my $genres= $request->getResult('genres_loop');
+			foreach my $genre (@$genres) {
+				$genreMatch ||= grep {uc($genre->{genre}) eq $_} split(/,/, uc($prefs->get('showComposerReleasesbyAlbumGenres')));
+				last if $genreMatch;
+			}
+		}
+
+		$_->{role_ids} =~ s/COMPOSER/COMPOSERALBUM/ if ( ( $menuMode && $menuMode ne 'artists' && $menuRoles ) || $genreMatch || $prefs->get('showComposerReleasesbyAlbum')==1 );
```

Comment:
A rule of thumb for inline conditions etc. is that you should use a regular multiline `if () { do something }` if the line was longer than 80-100 characters.

That said... I believe I don't like what you're doing there. I understand you want to change the role to a custom value given the specific condition. You do so to later on treat this particular case slightly differently. Is this correct? If so, couldn't we flag this slightly differently? `role_id` usually has a well defined set of values. If the manipulated value got propagated for whatever reason, some code elsewhere could get a problem.

Couldn't this be handled in the loop around line 120, where we have some "special treatment" for the `TRACKARTIST` too? That way we could store the data directly in `$contributions{COMPOSERALBUM}` instead of manipulating the `role_ids`.

Reply:
Makes sense. I'll work on this. Will it break things if I change `$role` within the loop? Or is `$role` purely local within the loop?

And yes, I know the condition at 83 has become unwieldy - I think I'll evaluate it separately and avoid retrieving the pref multiple times inside the albums loop.